src/share/index.ts

```diff
@@ -35,6 +35,9 @@ export type CreateLinkOptions = {
 
     /** Payload to include with shared link. */
     data?: unknown,
+
+    /** Create a link to this explicit URL. It must have the same origin as the current document. If not specified, the current URL will be used. */
+    url?: URL,
```

Comment:
Do we need this to be a `URL` object instead of a plain string?

```js
playpass.createLink({ url: new URL("/my/page", location.href) })
```

vs:

```js
playpass.createLink({ url: "/my/page" })
```

Also why the same-origin restriction?

Reply:
I debated this and thought ULR was easier to work with because I could just do 

let a = new URL(document.location)
a.search = ''
a.path = 'foo'

And pass that in and let the .href field do the string concat rather than me do it. 

Changing the origin feels like a footgun/mistake, so why not prevent it? Can't think of a legit use case for this.